src/telliot_ampl_feeds/sources/usd_vwap.py

```diff
@@ -0,0 +1,205 @@
+import asyncio
+import statistics
+from abc import ABC
+from dataclasses import dataclass
+from dataclasses import field
+from typing import Any
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Tuple
+from typing import TypeVar
+
+import requests
+from telliot.datasource import DataSource
+from telliot.types.datapoint import datetime_now_utc
+from telliot.types.datapoint import OptionalDataPoint
+from telliot.utils.response import ResponseStatus
+
+from telliot_ampl_feeds.config import AMPLConfig
+
+
+T = TypeVar("T")
+
+
+@dataclass
+class AMPLSource(DataSource):
```

Comment:
This is just an opinion, but class definition might be overkill here (inheritors don't get a lot).
Could just have a function def:
`async def get_float_from_api(url: str, params, headers=None)`
I'd like to see `store_datapoint(datapoint)` done explicitly in the `fetch_new_datapoint() `call anyway,
just for readability and what we will be asking people to do.

Reply:
It's also a little confusing to have a subclass of DataSource that does not immediately define `fetch_new_datapoint()`

Reply:
Good points, will change